allennlp/modules/elmo.py

```diff
@@ -515,3 +565,63 @@ def forward(self,  # pylint: disable=arguments-differ
                 'activations': output_tensors,
                 'mask': mask,
         }
+
+    def create_cached_cnn_embeddings(self, tokens: List[str]) -> None:
+        """
+        Given a list of tokens, this method precomputes word representations
+        by running just the character convolutions and highway layers of elmo,
+        essentially creating uncontextual word vectors. On subsequent forward passes,
+        if all of the words in the batch have cached, the word ids are looked up from
+        an embedding, rather than being computed on the fly via the CNN encoder.
+
+        This function sets 3 attributes:
+
+        _word_embedding : ``torch.Tensor``
+            The word embedding for each word in the tokens passed to this method.
+        _bos_embedding : ``torch.Tensor``
+            The embedding for the BOS token.
+        _eos_embedding : ``torch.Tensor``
+            The embedding for the EOS token.
+
+        Parameters
+        ----------
+        tokens : ``List[str]``, required.
+            A list of tokens to precompute character convolutions for.
+        """
+        tokens = [ELMoCharacterMapper.bos_token, ELMoCharacterMapper.eos_token] + tokens
+        timesteps = 32
+        batch_size = 32
+        chunked_tokens = lazy_groups_of(iter(tokens), timesteps)
+
+        all_embeddings = []
+        device = get_device_of(next(self.parameters()))
+        for batch in lazy_groups_of(chunked_tokens, batch_size):
+            # Shape (batch_size, 32, 32, 50)
```

Comment:
Isn't this `(batch_size, timesteps, 50)`, or `(32, 32, 50)`?  Or are there really 4 dimensions here?